pkgs/development/libraries/boost/generic.nix

```diff
@@ -64,6 +64,7 @@ let
     "link=${link}"
     "-sEXPAT_INCLUDE=${expat.dev}/include"
     "-sEXPAT_LIBPATH=${expat.out}/lib"
+    "define=BOOST_STACKTRACE_LIBCXX_RUNTIME_MAY_CAUSE_MEMORY_LEAK"
```

Comment:
This is a little scary. Do we feel okay about the purported memory leak?

Reply:
We disable the stacktrace from exception feature in the other line added in this commit in case `libcxx` is used. I don't understand why this has to be set when libstdc++ is used, but hopefully upstream will clarify (https://github.com/boostorg/stacktrace/issues/163#issuecomment-2495857604).

Reply:
I wonder if this is a 32‐bit thing specifically? I’ve been doing testing on `aarch64-linux` with no issues.

Reply:
Strange, I thought this would come up everywhere except on x86_64. See my other comment here: https://github.com/NixOS/nixpkgs/pull/356102#issuecomment-2495469525.

Reply:
It builds the `noop`, `basic`, and `addr2line` variants of `stacktrace`. Perhaps the fact that it’s skipping `backtrace` on `aarch64-linux` shields us?

```
…
    - libbacktrace builds      : no [2]
    - cxx11_rvalue_references  : yes [3]
    - libbacktrace builds      : no [3]
    - addr2line builds         : yes [2]
    - WinDbg builds            : no [2]
    - WinDbg builds            : no [3]
    - WinDbgCached builds      : no [2]
    - WinDbgCached builds      : no [3]
    - BOOST_COMP_GNUC >= 4.3.0 : no [2]
    - BOOST_COMP_GNUC >= 4.3.0 : no [4]
    - cxx11_hdr_thread         : yes [2]
    - cxx11_hdr_regex          : yes [2]

[1] gcc-14
[2] gcc-14/release/arm_64/threading-multi/visibility-hidden
[3] gcc-14/release/arm_64/build-no/threading-multi/visibility-hidden
[4] gcc-14/release/arm_64/link-static/threading-multi/visibility-hidden
```

Reply:
Ah yes, that must be it!

Should we make the define conditional then?

Reply:
I guess it’s harmless? But this also suggests that another option would be to just disable the `backtrace` variant on the problematic platforms; not sure if that would be better or not.

Reply:
I regrouped the code so now either one or the other config option gets passed and everything is in one place.